src/EventStore/LaravelEventStore.php

```diff
@@ -95,27 +99,23 @@ public function append($id, DomainEventStream $eventStream)
     }
 
     /**
-     * @param DomainMessage $domainMessage
+     * @param array $eventRow
+     * @param bool $ignorePlayhead
+     * @throws DuplicateAggregatePlayhead
      */
-    private function insertEvent(DomainMessage $domainMessage)
+    private function insertEvent(array $eventRow, bool $ignorePlayhead)
     {
         try {
-            $this->db->table($this->eventStoreTableName)
-                     ->insert(
-                         [
-                             'uuid'        => (string)$domainMessage->getId(),
-                             'playhead'    => $domainMessage->getPlayHead(),
-                             'metadata'    => json_encode($this->serializer->serialize($domainMessage->getMetadata())),
-                             'payload'     => json_encode($this->serializer->serialize($domainMessage->getPayload())),
-                             'recorded_on' => (string)$domainMessage->getRecordedOn(),
-                             'type'        => $domainMessage->getType(),
-                         ]
-                     );
+            $this->db->table($this->eventStoreTableName)->insert($eventRow);
         } catch (\PDOException $ex) {
-            if ($ex->getCode() == 23000) {
-                throw new DuplicateAggregatePlayhead((string)$domainMessage->getId(), $domainMessage->getPlayHead());
+            if ((string) $ex->getCode() === '23000') {
+                if ($ignorePlayhead) {
+                    $eventRow['playhead'] ++;
```

Comment:
In all seriousness though, I am not sure. Maybe try a couple of times and then exception?